driver/reducecalculation_driver.hpp

```diff
@@ -128,17 +181,20 @@ class ReduceCalculationDriver : public Driver
 private:
     InputFlags inflags;
 
-    miopenTensorDescriptor_t inputDesc;
-    miopenTensorDescriptor_t outputDesc;
+    miopenTensorDescriptor_t inputDesc  = nullptr;
+    miopenTensorDescriptor_t outputDesc = nullptr;
 
-    std::unique_ptr<GPUMem> in_dev;
-    std::unique_ptr<GPUMem> out_dev;
-    std::unique_ptr<GPUMem> workspace_dev;
+    std::unique_ptr<GPUMem> in_dev        = nullptr;
+    std::unique_ptr<GPUMem> out_dev       = nullptr;
+    std::unique_ptr<GPUMem> workspace_dev = nullptr;
 
     std::vector<Tgpu> in;
     std::vector<Tgpu> out;
     std::vector<Tref> outhost;
 
+    std::vector<uint8_t> logical_out;
+    std::vector<uint8_t> logical_outhost;
+
     size_t ws_sizeInBytes;
 
     int dim;
```

Comment:
change to `uint32_t`

Reply:
The output is intended to be of type `bool`. However, MIOpen does not yet support `TensorDescription` with the `bool` type (e.g., `MIOpenBool`). As a workaround, I used `uint8_t`. Could you please explain why you suggest using `uint32_t` instead?

Reply:
oh it was mistake, I suggested changing dim to `uint32_t`